amazon-kinesis-client/src/test/java/software/amazon/kinesis/config/KCLAppConfig.java

```diff
@@ -19,48 +29,92 @@
  * Consumer: streaming configuration (vs polling) that starts processing records at shard horizon
  */
+@Slf4j
 public abstract class KCLAppConfig {
+    public static final String AWS_ACCOUNT_PROFILE_PROPERTY = "awsProfile";
+    public static final String CROSS_ACCOUNT_PROFILE_PROPERTY = "awsCrossAccountProfile";
+    public static final String CROSS_ACCOUNT_CONSUMER_NAME = "cross-account-consumer";
+    public static final String INTEGRATION_TEST_RESOURCE_PREFIX = "KCLIntegrationTest";
 
-    private KinesisAsyncClient kinesisAsyncClient;
+    private String accountIdForConsumer = null;
+    private String accountIdForStreamOwner = null;
+    private List<String> streamNames = null;
+    private KinesisAsyncClient kinesisAsyncClientForConsumer;
+    private StsAsyncClient stsAsyncClientForConsumer;
+    private KinesisAsyncClient kinesisAsyncClientForStreamOwner;
+    private StsAsyncClient stsAsyncClientForStreamOwner;
     private DynamoDbAsyncClient dynamoDbAsyncClient;
     private CloudWatchAsyncClient cloudWatchAsyncClient;
     private RecordValidatorQueue recordValidator;
 
     /**
-     * Name used for test stream and lease tracker table
+     * List of Strings, either stream names or valid stream Arns, to be used in testing. For single stream mode, return
+     * a list of size 1. For multistream mode, return a list of size > 1.
      */
-    public abstract String getStreamName();
+    public abstract List<Arn> getStreamArns();
+
+    public List<String> getStreamNames() {
+        if (this.streamNames == null) {
+            return getStreamArns().stream().map(streamArn ->
+                                          streamArn.toString().substring(streamArn.toString().indexOf("/") + 1))
+                                  .collect(Collectors.toList());
+        } else {
+            return this.streamNames;
+        }
+    }
+
+    public abstract String getTestName();
+
+    public String getApplicationName() {
+        return INTEGRATION_TEST_RESOURCE_PREFIX + getTestName();
+    }
 
     public int getShardCount() { return 4; }
 
     public Region getRegion() { return Region.US_WEST_2; }
 
     /**
-     * "default" profile, should match with profiles listed in "cat ~/.aws/config"
+     * Gets credentials for passed in profile with "-DawsProfile" which should match "~/.aws/config". Otherwise,
+     * uses default profile credentials chain.
      */
     private AwsCredentialsProvider getCredentialsProvider() {
-        final String awsProfile = System.getProperty("awsProfile");
+        final String awsProfile = System.getProperty(AWS_ACCOUNT_PROFILE_PROPERTY);
         return (awsProfile != null) ?
                 ProfileCredentialsProvider.builder().profileName(awsProfile).build() : DefaultCredentialsProvider.create();
     }
 
+    public boolean isCrossAccount() {
+        return false;
+    }
+
+    public AwsCredentialsProvider getCrossAccountCredentialsProvider() {
+        return null;
+    }
```

Comment:
would it be cleaner for this to be a `protected` method called `getCredentialsProviderForStreamOwner`?
in `KCLAppConfig` class, this can provide the same credentials as the `getCredentialsProvider` method
then the inherited `KCLCrossAccountAppConfig` class can override this `getCredentialsProviderForStreamOwner` method with different credentials

might be slightly more intuitive and simplify some of the other logic such as in `buildAsyncKinesisClientForStreamOwner`?